lib/latest/RabbitMQ.js

```diff
@@ -207,6 +207,24 @@ class RabbitMQ {
     }
   }
 
+  /**
+   * Unsubscribe from a queue or topic
+   * @param {Object} ch queue channel
+   * @param {String} consumerTag consumer identification
+   * @returns {Promise} resolves on success
+   * @public
+   */
+  async unsubscribe({
+    ch, consumerTag,
+  }) {
+    assert(ch, 'Missing channel object');
+    assert(consumerTag, 'Missing consumer tag');
+    await ch.cancel(consumerTag, (err) => {
+      if (err) {
+        this.debug(err);
+      }
+    });
```

Comment:
hi,

* please wrap this command in a try/catch block to capture the error returned by ch.cancel, ex:
```
try {
  ssert(ch, 'Missing channel object');
  await ch.cancel(consumerTag);
} catch (err) {
  this.debug(err);
  throw Reflect.construct(IError, [{
        name:     'MQ_UNSUBSCRIBE_ERROR',
        source:   `${this.sourceIdentifier}.unsubscribe`,
        message:  errMsg,
        severity: 'WARNING',
        cause:    err,
      }]);
}
```
* by providing a callback to cancel method it will not behave like a promise
* from where do you obtain reference to channel ?
* there is no test for this method

Thank you,

Reply:
Hi,
- I'll wrap it as instructed
- agree
- new code has that
- test added, but I'll modify it for new code.
Best